metadata-ingestion/src/datahub/ingestion/source/sql/sql_common.py

```diff
@@ -669,12 +693,14 @@ def loop_tables(  # noqa: C901
 
                 try:
                     yield from self._process_table(
-                        dataset_name, inspector, schema, table, sql_config
+                        dataset_name, inspector, schema, table, sql_config, data_reader
                     )
                 except Exception as e:
                     self.warn(logger, f"{schema}.{table}", f"Ingestion error: {e}")
         except Exception as e:
             self.error(logger, f"{schema}", f"Tables error: {e}")
+        if data_reader:
+            data_reader.close()
```

Comment:
I generally prefer to use context managers instead of explicitly calling .close() - we have a `Closeable` helper class

then we can use this structure, and close will automatically get called

```
with (data_reader or contextlib.nullcontext()):
	# rest of code
```

Reply:
Thank you ! I was looking for something like this.